addOns/automation/src/main/java/org/zaproxy/addon/automation/AutomationStatisticTest.java

```diff
@@ -0,0 +1,97 @@
+/*
+ * Zed Attack Proxy (ZAP) and its related class files.
+ *
+ * ZAP is an HTTP/HTTPS proxy for assessing web application security.
+ *
+ * Copyright 2021 The ZAP Development Team
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.zaproxy.addon.automation;
+
+import java.util.Arrays;
+import org.parosproxy.paros.Constant;
+import org.parosproxy.paros.control.Control;
+import org.zaproxy.zap.extension.stats.ExtensionStats;
+
+public class AutomationStatisticTest {
+    public final String key;
+    public final String name;
+    public final String operator;
+    public final long value;
+    public final String action;
+
+    public AutomationStatisticTest(
+            String key, String name, String operator, long value, String action)
+            throws IllegalArgumentException {
+        if (!Arrays.asList("warn", "error", "ignore").contains(action)) {
+            throw new IllegalArgumentException(
+                    Constant.messages.getString("automation.tests.stats.invalidAction", action));
+        }
+        if (!Arrays.asList("<", ">", "<=", ">=", "==", "!=").contains(operator)) {
+            throw new IllegalArgumentException(
+                    Constant.messages.getString(
+                            "automation.tests.stats.invalidOperator", operator));
+        }
+        this.key = key;
+        this.name = name;
+        this.operator = operator;
+        this.value = value;
+        this.action = action;
+    }
+
+    public void logToProgressIfFailed(AutomationProgress progress) throws RuntimeException {
+        if (!hasFailed()) {
+            return;
```

Comment:
Log to info that its passed? Lets the user know the test was valid and may give additional useful feedback if we state the exact result, eg something like `Job myjob Test passed stats.aaa.bbb 123 >= 100`